data/src/test/test_ndvi.py

```diff
@@ -0,0 +1,237 @@
+            dummy_df["ndvi_one_year_change_mean"] = [0.1, 0.0, -0.2]
+            mock_extract.return_value = dummy_df
+            results = extract_ndvi_to_parcels(self.test_raster_path, self.test_parcels)
+            self.assertEqual(len(results), 3)
+            self.assertIn("ndvi_mean", results.columns)
+            self.assertIn("ndvi_one_year_change_mean", results.columns)
+            for idx in results.index:
+                self.assertGreaterEqual(results.loc[idx, "ndvi_mean"], -0.2)
+                self.assertLessEqual(results.loc[idx, "ndvi_mean"], 0.8)
+
+    def test_extract_ndvi_to_parcels_reprojection(self):
+        parcels_use_crs = self.test_parcels.to_crs(USE_CRS)
+        with patch("rioxarray.open_rasterio") as mock_open:
+            mock_dataset = MagicMock()
+            mock_dataset.rio.crs = "EPSG:4326"
+            mock_dataset.rio.reproject.return_value = mock_dataset
+            mock_dataset.rio.to_raster.return_value = None
+            mock_open.return_value.__enter__.return_value = mock_dataset
+            with patch("exactextract.exact_extract") as mock_extract:
+                mock_extract.return_value = gpd.GeoDataFrame(
+                    {
+                        "id": [1, 2, 3],
+                        "ndvi_mean": [0.5, 0.7, -0.1],
+                        "ndvi_one_year_change_mean": [0.0, 0.1, -0.2],
+                    }
+                )
+                extract_ndvi_to_parcels(self.test_raster_path, parcels_use_crs)
+                mock_dataset.rio.reproject.assert_called_once_with(USE_CRS)
+                mock_dataset.rio.to_raster.assert_called_once()
+
+    def test_extract_ndvi_invalid_raster(self):
+        with self.assertRaises(FileNotFoundError):
+            extract_ndvi_to_parcels("nonexistent_raster.tif", self.test_parcels)
+
+    @patch("new_etl.data_utils.ndvi.odc.stac.stac_load")
+    @patch("new_etl.data_utils.ndvi.get_current_summer_year")
+    @patch("new_etl.data_utils.ndvi.pystac_client.Client.open")
+    def test_generate_ndvi_data_failure(
+        self, mock_client_open, mock_get_year, mock_stac_load
+    ):
+        mock_get_year.return_value = 2025
+        dummy_search = MagicMock()
+        dummy_search.item_collection.return_value = ["dummy_item"]
+        dummy_client = MagicMock()
+        dummy_client.search.return_value = dummy_search
+        mock_client_open.return_value = dummy_client
+        mock_stac_load.side_effect = Exception("Test exception in stac_load")
+        with self.assertRaises(Exception) as context:
+            generate_ndvi_data(
+                os.path.join(self.temp_dir, "dummy.tif"),
+                self.test_polygon,
+                (-75.2, 39.9, -75.1, 40.0),
+            )
+        self.assertIn("Test exception", str(context.exception))
+
+    @patch("tempfile.gettempdir")
+    @patch("new_etl.data_utils.ndvi.get_current_summer_year")
+    @patch("new_etl.data_utils.ndvi.get_bbox_from_census_data")
```

Comment:
it is hard to reason about tests when so much is patched. See if we can eliminate at least 2 of the patches from my suggestions.